google/cloud/storage/tests/object_insert_integration_test.cc

```diff
@@ -118,6 +117,33 @@ TEST_F(ObjectInsertIntegrationTest, XmlInsertWithMD5) {
   client.DeleteObject(bucket_name, object_name);
 }
 
+TEST_F(ObjectInsertIntegrationTest, InsertWithMetadata) {
+  Client client;
+  auto bucket_name = ObjectTestEnvironment::bucket_name();
+  auto object_name = MakeRandomObjectName();
+
+  std::string expected = LoremIpsum();
+
+  // Create the object, but only if it does not exist already.
+  ObjectMetadata meta = client.InsertObject(
+      bucket_name, object_name, expected, IfGenerationMatch(0),
+      WithObjectMetadata(ObjectMetadata()
+                             .upsert_metadata("test-key", "test-value")
+                             .set_content_type("text/plain")));
```

Comment:
Wow. I kind of like it because I'm not forced to use `ObjectMetadata()` with no metadata set.
Are you thinking of doing this to for other requests?

Reply:
That is the plan.  `UploadFile()`, `WriteObject()`, `CopyObject()`, `ComposeObject()`, `RewriteObject()`, etc. should get this treatment.